contracts/vaults/STETHVault/base/RibbonVault.sol

```diff
@@ -0,0 +1,685 @@
+    }
+
+    /**
+     * @notice Initiates a withdrawal that can be processed once the round completes
+     * @param shares is the number of shares to withdraw
+     */
+    function initiateWithdraw(uint128 shares) external nonReentrant {
+        require(shares > 0, "!shares");
+
+        // We do a max redeem before initiating a withdrawal
+        // But we check if they must first have unredeemed shares
+        if (
+            depositReceipts[msg.sender].amount > 0 ||
+            depositReceipts[msg.sender].unredeemedShares > 0
+        ) {
+            _redeem(0, true);
+        }
+
+        // This caches the `round` variable used in shareBalances
+        uint256 currentRound = vaultState.round;
+        Vault.Withdrawal storage withdrawal = withdrawals[msg.sender];
+
+        bool topup = withdrawal.round == currentRound;
+
+        emit InitiateWithdraw(msg.sender, shares, currentRound);
+
+        uint256 withdrawalShares = uint256(withdrawal.shares);
+
+        if (topup) {
+            uint256 increasedShares = withdrawalShares.add(shares);
+            ShareMath.assertUint128(increasedShares);
+            withdrawals[msg.sender].shares = uint128(increasedShares);
+        } else if (withdrawalShares == 0) {
+            withdrawals[msg.sender].shares = shares;
+            withdrawals[msg.sender].round = uint16(currentRound);
+        } else {
+            // If we have an old withdrawal, we revert
+            // The user has to process the withdrawal
+            revert("Existing withdraw");
+        }
+
+        vaultState.queuedWithdrawShares = uint128(
+            uint256(vaultState.queuedWithdrawShares).add(shares)
+        );
+
+        _transfer(msg.sender, address(this), shares);
+    }
+
+    /**
+     * @notice Completes a scheduled withdrawal from a past round. Uses finalized pps for the round
+     * @param minETHOut is the min amount of `asset` to recieve for the swapped amount of steth in crv pool
+     */
+    function completeWithdraw(uint256 minETHOut) external nonReentrant {
+        Vault.Withdrawal storage withdrawal = withdrawals[msg.sender];
+
+        uint256 withdrawalShares = withdrawal.shares;
+        uint256 withdrawalRound = withdrawal.round;
+
+        // This checks if there is a withdrawal
+        require(withdrawalShares > 0, "!initiated");
+
+        require(withdrawalRound < vaultState.round, "Round not closed");
+
+        // We leave the round number as non-zero to save on gas for subsequent writes
+        withdrawals[msg.sender].shares = 0;
+        vaultState.queuedWithdrawShares = uint128(
+            uint256(vaultState.queuedWithdrawShares).sub(withdrawalShares)
+        );
+
+        uint256 withdrawAmount =
+            ShareMath.sharesToUnderlying(
+                withdrawalShares,
+                roundPricePerShare[uint16(withdrawalRound)],
+                vaultParams.decimals
+            );
+
+        // Unwrap may incur curve pool slippage
+        uint256 amountETHOut =
```

Comment:
the amount we transfer takes into account curve pool slippage - so they will get less. 

same for withdrawInstantly()